sklearn/ensemble/_voting.py

```diff
@@ -273,6 +274,28 @@ def __init__(
         self.flatten_transform = flatten_transform
         self.verbose = verbose
 
+    def _validate_estimators(self):
```

Comment:
I think we can simplify this implementation:

```python
   def _validate_estimators(self):
        names, clfs = super()._validate_estimators()

        updated_clfs = []
        for clf in clfs:
            params_with_class_weights = [
                (k, v)
                for k, v in clf.get_params(deep=True).items()
                if k.endswith("class_weight") and isinstance(v, Mapping)
            ]
            # re-encode params with class weights
            if params_with_class_weights:
                clf = clone(clf)
                for k, v in params_with_class_weights:
                    labels, weights = zip(*v.items())
                    encoded_labels = self.le_.transform(labels)
                    new_weights = dict(zip(encoded_labels, weights))
                    clf.set_params(**{k: new_weights})

            updated_clfs.append(clf)
        return names, tuple(updated_clfs)
```

This removes the `already_clone` and reduces the number of indentations. Also we can use `self.le_` to do the transformation.